bin/geni-sync-wireless

```diff
@@ -0,0 +1,618 @@
+    #     membership in ORBIT groups
+    # Make sure project lead on wimax-enabled projects translates to
+    #     admin in ORBIT group
+    # If we're doing cleanup:
+    #    Delete group members not in project
+    #    Delete groups not corresponding to projects
+    #    Disable users in no group (other than holdingpen)
+    def synchronize(self):
+
+        now = datetime.datetime.now()
+        print "Synchronizing GENI wimax-enabled projects/users with ORBIT: %s"\
+            % datetime.datetime.strftime(now, '%Y-%m-%d %H:%M:%S')
+
+        # Grab project info for GENI wimax-enabled projects
+        # Filtering to given project if set with --project
+        self.get_geni_projects()
+
+        # Grab members in wimax-enabled projects
+        self.get_geni_members()
+
+        # Remove disabled members from projects and members 
+        # (unless member is lead)
+        self.remove_disabled_members()
+
+        # Get the ORBIT list of groups and admins
+        self._orbit_groups, self._orbit_users = \
+            self._orb.get_orbit_groups_and_users()
+        if self._options.verbose:
+            print "GENI PROJECTS = %s" % self._geni_projects
+            print "GENI MEMBERS = %s" % self._geni_members
+            print "ORBIT GROUPS = %s" % self._orbit_groups
+            print "ORBIT USERS = %s" % self._orbit_users
+
+        # Make sure the holdingpen gorup and admin exist
+        self.ensure_holdingpen_group_and_admin()
+
+        # Make sure all members of wimax-enabled projects exist as orbit users
+        # Make sure they are enabled
+        self.ensure_project_members_exist()
+
+        # Make sure all wimax-enabled projects exist as orbit groups
+        self.ensure_projects_exist()
+
+        # Make sure all orbit users are in proper wimax group
+        self.ensure_project_members_in_groups()
+
+        # Make sure the admins of orbit groups match the leads of GENI projects
+        self.ensure_project_leads_are_group_admins()
+
+        # If we're doing cleanup, 
+        #   delete group members who aren't project members
+        #   delete groups that aren't GENI projects
+        #   disable any users not in any GENI project
+        if self._cleanup:
+            self.delete_group_members_not_in_project()
+            self.delete_groups_without_projects()
+            self.disable_users_in_no_project()
+
+    # Make sure that the holdingpen group exists
+    def ensure_holdingpen_group_and_admin(self):
+
+        # Find the holdingpen admin among the GENI members read
+        holdingpen_admin_info = None
+        for member_id, member_info in self._geni_members.items():
+            if member_info['username'] == self._options.holdingpen_admin:
+                holdingpen_admin_info = member_info
+                break
+
+        if not holdingpen_admin_info:
+            self.error("Holdingpen admin not in GENI: %s" % \
+                           self._options.holdingpen_admin)
```

Comment:
Actually self.error prints a message to syslog and then exists, so this won't fall through.